### ISSUE TITLE
mrgingham currently only accepting 10x10 grid

### ISSUE THREAD

**Issue by Zepheoss** [dkogan/mrcal#29]
Does a new update of mrgingham only allow 10x10 grids? I currently have a 5x7 checkerboard that worked in the past, wondering if I need to make a new one. 

When I run:
mrcal-calibrate-cameras --corners-cache corners.vnl --lensmodel LENSMODEL_OPENCV8 --focal 766.67 --object-spacing 0.04 --object-width-n 5 '*.png'

I get the exception:
mrgingham currently accepts ONLY 10x10 grids

Currently running mrcal 2.1-5build1

Thanks for the help!

**Comment by dkogan** [comment]
Hello. mrgingham has never accepted non-square grids, and I'm sure your 5x7 never really worked properly.

This would be a nice thing to add, but it's a bit non-trivial, and is very low priority for me, so realistically, this will never happen. If you really need to use non-square grids, try a different detector. BoofCV works pretty well.

**Comment by dkogan** [comment]
Since this probably will never be fixed, I'm going to close the issue. Sorry.

**Comment by Zepheoss** [comment]
Thank you for the reply, your work is instrumental to my project.

I should have specified more clearly, does mrgingham not allow smaller than 10x10 grids that are still square?

**Comment by dkogan** [comment]
Smaller than 10x10 grids that are still square should work. It's something I've very rarely actually tried, though. Let me know if that gives you trouble.

**Comment by Zepheoss** [comment]
Thank you for pointing out that my grid isn't symmetric, I'll be getting one that is square in the near future. 

Basically, since my project is using low resolution LWIR cameras, I have to use bigger checkers on the checkerboard for the camera to see. A 5x5 grid is the most feasible to hold and image in various rotations and distances for this. My problem is that the command I pasted in my first post will not even run unless --object-width is set to 10.     

To debug I took screenshots of a couple different square checkerboards with various widths, the only one that mrgingham would accept is one with width of 10. The smaller checkerboards would not work since --object-width would have to be set to less than 10. The same goes for a checkerboard with width of 14 with --object-width set also to 14. 

I'm sure it has something to do with that I am doing since you have examples on your site of using larger than 10x10 grids. I'm just at a loss on why mrgingham throwing this exception for me. 

Thanks for the support, it's greatly appreciated.
